polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -2143,8 +2141,34 @@ MLIRASTConsumer::getOrCreateGlobal(const ValueDecl &VD, std::string Prefix,
     break;
   }
 
-  // Initialize the global variable.
-  if (const Expr *Init = Var->getInit()) {
+  // Initialize the global.
+  const VarDecl *InitDecl;
+  const Expr *InitExpr = Var->getAnyInitializer(InitDecl);
+
+  if (!InitExpr) {
+    if (DefKind == VarDecl::TentativeDefinition) {
```

Comment:
I'd say this is the culprit of the issue @whitneywhtsang mentions. I'm no C expert, but changing the extension of the updated tests from `.c` to `.cpp` makes some global variables `uninitialized`. Maybe we should differ between C and C++ here?

Reply:
I fixed the C++ problems now